middle_end/flambda/terms/continuation_handlers.rec.ml

```diff
@@ -46,14 +46,20 @@ let apply_name_permutation t perm =
     Continuation.Map.empty
 
 let all_ids_for_export t =
-  Continuation.Map.fold (fun _k handler ids ->
+  Continuation.Map.fold (fun k handler ids ->
       Ids_for_export.union ids
-        (Continuation_handler.all_ids_for_export handler))
+        (Ids_for_export.add_continuation
+          (Continuation_handler.all_ids_for_export handler)
+          k))
     t
     Ids_for_export.empty
 
 let import import_map t =
-  Continuation.Map.map (Continuation_handler.import import_map) t
+  Continuation.Map.of_list
+    (List.map (fun (k, handler) ->
+        Ids_for_export.Import_map.continuation import_map k,
+        Continuation_handler.import import_map handler)
+      (Continuation.Map.bindings t))
```

Comment:
This seems like it would be more efficient if it used `Continuation.Map.fold` to avoid the lists.

Reply:
What if I used `Seq.map` instead?

Reply:
Might be ok, but you might want to check what `Seq` does internally first...

Reply:
Doesn't seem to do anything crazy, just applies the function to each element as it's generated. In particular, it won't build up any intermediate structure.